go-controller/pkg/ovn/controller/apbroute/external_controller_pod.go

```diff
@@ -28,6 +30,13 @@ func (m *externalPolicyManager) syncPod(pod *corev1.Pod, routeQueue workqueue.Ty
 }
 
 func getExGwPodIPs(gatewayPod *corev1.Pod, networkName string) (sets.Set[string], error) {
+	// If an external gateway pod is in terminating or not ready state then don't return the
+	// IPs for the external gateway pod
+	if util.PodTerminating(gatewayPod) || util.PodNotReady(gatewayPod) {
```

Comment:
i think this check should happen before getIPs is called? i.e similar to how its done inside addPodExternalGW ?
we shouldn't even call getIPs if we know we don't want to call it